TheMovieDbTests/HomeViewModelTests.swift

```diff
@@ -0,0 +1,168 @@
+    }
+    
+    func test_mockGetMoviesUpcoming() {
+        HomeViewModelTests.resource = "Upcoming"
+        let category = Category(name: HomeViewModelTests.resource, path: MovieDbEndPoints.trendingUrl, movies: [])
+        viewModel = HomeViewModel(client: MockMovieClient(), categories: [category])
+        
+        let expectation = expectation(description: "Expected categories")
+        viewModel.isLoading = {
+            
+            XCTAssertFalse(self.viewModel.categories.isEmpty)
+            XCTAssertEqual(self.viewModel.categories.count, 1)
+            XCTAssertEqual(self.viewModel.categories.first?.movies.count, 20)
+            XCTAssertEqual(self.viewModel.categories.first?.movies.first?.title, "Venom: Let There Be Carnage")
+            expectation.fulfill()
+        }
+        
+        viewModel.getMovies()
+        
+        waitForExpectations(timeout: 1.0, handler: nil)
+    }
+    
+    func test_mockGetMoviesNowPlaying() {
+        HomeViewModelTests.resource = "NowPlaying"
+        let category = Category(name: HomeViewModelTests.resource, path: MovieDbEndPoints.trendingUrl, movies: [])
+        viewModel = HomeViewModel(client: MockMovieClient(), categories: [category])
+       
+        let expectation = expectation(description: "Expected categories")
+        viewModel.isLoading = {
+            
+            XCTAssertFalse(self.viewModel.categories.isEmpty)
+            XCTAssertEqual(self.viewModel.categories.count, 1)
+            XCTAssertEqual(self.viewModel.categories.first?.movies.count, 20)
+            XCTAssertEqual(self.viewModel.categories.first?.movies.first?.title, "Venom: Let There Be Carnage")
+            expectation.fulfill()
+        }
+        
+        viewModel.getMovies()
+        
+        waitForExpectations(timeout: 1.0, handler: nil)
+    }
+    
+    func test_mockGetMoviesPopular() {
+        HomeViewModelTests.resource = "Popular"
+        let category = Category(name: HomeViewModelTests.resource, path: MovieDbEndPoints.trendingUrl, movies: [])
+        viewModel = HomeViewModel(client: MockMovieClient(), categories: [category])
+       
+        let expectation = expectation(description: "Expected categories")
+        viewModel.isLoading = {
+            
+            XCTAssertFalse(self.viewModel.categories.isEmpty)
+            XCTAssertEqual(self.viewModel.categories.count, 1)
+            XCTAssertEqual(self.viewModel.categories.first?.movies.count, 20)
+            XCTAssertEqual(self.viewModel.categories.first?.movies.first?.title, "Venom: Let There Be Carnage")
+            expectation.fulfill()
+        }
+        
+        viewModel.getMovies()
+        
+        waitForExpectations(timeout: 1.0, handler: nil)
+    }
+    
+    func test_mockGetMoviesError() {
+        HomeViewModelTests.resource = "Error"
+        let category = Category(name: HomeViewModelTests.resource, path: MovieDbEndPoints.trendingUrl, movies: [])
```

Comment:
I can't run the tests because there is an error in this line about a missing parameter, please check it out.

Reply:
Got it.